### PR TITLE
Rails 7 upgrade from 6.1.7.7 to 7.0.8.1

### PR DESCRIPTION
**Pull request by GcioGregg**
## Description
Upgrade Rails to 7.0.8.1 from 6.1.7.7

## Original issue(s)
[vfep-1265](https://jira.devops.va.gov/browse/VFEP-1265)

## Testing done
Developer user testing of GIBCT dashboard and Comparison Tool on local sandbox.  All RSpec & Rubocop tests pass.

## Acceptance criteria
- [x] GIBCT Dashboard and Comparison Tool work as expected.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: config/environments/production.rb [hunk 1]
```diff
@@ -14,17 +16,17 @@
...
+  # Ensures that a master key has been made available in either ENV["RAILS_MASTER_KEY"]
+  # or in config/master.key. This key is used to decrypt credentials (and other encrypted files).
+  # config.require_master_key = true
 
   # Disable serving static files from the `/public` folder by default since
   # Apache or NGINX already handles this.
   config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
 
   # Compress JavaScripts and CSS.
+  # Do we want to switch to the Terser gem?
```
Comment (ryan-mcneil) [inline config/environments/production.rb:28]: Is this a question for reviewers, or just an idea for future work? Maybe create a ticket vs adding comments to a file that may be forgotten about and lose context?

File: config/environments/development.rb [hunk 2]
```diff
@@ -10,24 +12,54 @@
...
   # This option may cause significant delays in view rendering with a large
   # number of complex assets.
   config.assets.debug = true
 
+  # Raises error for missing translations.
+  # config.i18n.raise_on_missing_translations = true
+
+  # Annotate rendered view with file names.
+  # config.action_view.annotate_rendered_view_with_filenames = true
+
```
Comment (ryan-mcneil) [inline config/environments/development.rb:62]: Is there a reason for adding these commented-out settings?

File: config/environments/production.rb [hunk 3]
```diff
@@ -41,6 +49,16 @@
...
+  # https://guides.rubyonrails.org/active_storage_overview.html
+  # Store uploaded files on the local file system (see config/storage.yml for options).
+  # Disable https://stackoverflow.com/questions/49813214/disable-active-storage-in-rails-5-2
+  # config.active_storage.service = :local
+
+  # Mount Action Cable outside main process or domain.
+  # config.action_cable.mount_path = nil
+  # config.action_cable.url = "wss://example.com/cable"
+  # config.action_cable.allowed_request_origins = [ "http://example.com", /http:\/\/example.*/ ]
+
```
Comment (ryan-mcneil) [inline config/environments/production.rb:61]: more commented out settings?

File: config/environments/production.rb [hunk 4]
```diff
@@ -52,30 +70,34 @@
...
-
   # Use a different cache store in production.
   # config.cache_store = :mem_cache_store
 
-  # Enable serving of images, stylesheets, and JavaScripts from an asset server.
-  # config.action_controller.asset_host = 'http://assets.example.com'
-
+  # Use a real queuing backend for Active Job (and separate queues per environment).
+  # config.active_job.queue_adapter     = :resque
+  # config.active_job.queue_name_prefix = "gibct_data_service_production"
```
Comment (ryan-mcneil) [inline config/environments/production.rb:78]: and more

File: config/environments/production.rb [hunk 5]
```diff
@@ -14,17 +16,17 @@
...
   config.action_controller.perform_caching = true
 
-  # Enable Rack::Cache to put a simple HTTP cache in front of your application
-  # Add `rack-cache` to your Gemfile before enabling this.
-  # For large-scale production use, consider using a caching reverse proxy like
-  # NGINX, varnish or squid.
-  # config.action_dispatch.rack_cache = true
+  # Ensures that a master key has been made available in either ENV["RAILS_MASTER_KEY"]
+  # or in config/master.key. This key is used to decrypt credentials (and other encrypted files).
+  # config.require_master_key = true
```
[the remark this answers is not in the document]
Comment (nfstern02) [reply config/environments/production.rb:21]: It's commented out.

File: config/environments/production.rb (hunk 1 above)
Comment (nfstern02) [reply config/environments/production.rb:28]: It's a question based on stuff that popped up when I googled an issue I was having with this. What we have also works but some people recommended switching to this gem as an alternative.

File: config/environments/development.rb (hunk 2 above)
Comment (nfstern02) [reply config/environments/development.rb:62]: It's part of what got added when running bin/rails app:update. We're not using but I thought they should be left in.

File: config/environments/production.rb (hunk 3 above)
Comment (nfstern02) [reply config/environments/production.rb:61]: It's part of what got added when running bin/rails app:update. We're not using but I thought they should be left in.

File: config/environments/production.rb (hunk 4 above)
Comment (nfstern02) [reply config/environments/production.rb:78]: It's part of what got added when running bin/rails app:update. We're not using but I thought they should be left in.

File: config/environments/production.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (ryan-mcneil) [reply config/environments/production.rb:21]: oh haha, there were so many comments, I forgot what real code looks like :)

**Review by ryan-mcneil — APPROVED** [review]
Looks good to me. I don't LOVE all of the additional comments from the upgrade, but I suppose it's not a big deal if they're kept for future reference

**Review by nfstern02 — APPROVED** [review]

**Pull request merged**